tests/integration/components/input-test.js

```diff
@@ -59,4 +59,16 @@ module('Integration | Component | input', function(hooks) {
     assert.dom('input').hasClass('form-input');
     assert.dom('input').hasAttribute('role', 'searchbox');
   });
+
+  test('it ignores handlebars comments', async function(assert) {
+    await render(hbs`<Input {{! template-lint-disable-tree }} />`);
+
+    assert.dom('input').exists();
+  });
+
+  test('it ignores unknown attributes', async function(assert) {
```

Comment:
This test doesn't seem quite correct, the assertion doesn't check to confirm that `data-test-foo` or `aria-labelledby` are missing on the `input`. 

Also, once you update the tests to do those assertions you'll need to guard the test for <= 3.11 (because as of 3.11 this **does** work due to built-in support).

Reply:
Yeah, I thought about this. And agree the test is not that useful. Ideally it would assert that we see a warning, but as that happens at build-time, I don't see an easy way to test that.

I was mainly writing that test to ensure that passing unsupported attributes at least does not blow up, and to see the warning when running it locally on my machine.

Adding an assertion that the attribute is missing feels awkward to me tbh, as ideally we would support arbitrary attributes, just as native angle brackets do (if we had a good way to polyfill that). So seeing tests as a way to formally describe the features of our piece of software, in this case we would not describe and test for a feature, but a *missing feature*, right? Which seems odd. Or in other words, should we find a way to polyfill support for arbitrary attributes, that added feature should not break any tests!

Just my 2¢ though...

Reply:
> I was mainly writing that test to ensure that passing unsupported attributes at least does not blow up, and to see the warning when running it locally on my machine.

Ya, this makes sense.

> So seeing tests as a way to formally describe the features of our piece of software, in this case we would not describe and test for a feature, but a missing feature, right? Which seems odd.

Totally agree.